src/services/common/filesystem/fileCacheManager.spec.ts

```diff
@@ -0,0 +1,197 @@
+import fileSystemService from '@internxt-mobile/services/FileSystemService';
+import { ReadDirItem } from 'react-native-fs';
+import { FileCacheManager, FileCacheManagerConfig } from './fileCacheManager';
+
+jest.mock('@internxt-mobile/services/FileSystemService', () => {
+  return {
+    exists: jest.fn(),
+    readDir: jest.fn(),
+    getDirSize: jest.fn(),
+    statRNFS: jest.fn(),
+    moveFile: jest.fn(),
+    touch: jest.fn(),
+    unlinkIfExists: jest.fn(),
+  };
+});
+
+const mockedFs = jest.mocked(fileSystemService);
+const fileCacheManagerFactory = (config?: Partial<FileCacheManagerConfig>) =>
+  new FileCacheManager({
+    directory: '/cache',
+    maxFileSizeToCacheInBytes: 1024 * 1024 * 50,
+    maxSpaceInBytes: 1024 * 1024 * 500,
+    ...config,
+  });
+describe('File Cache Manager', () => {
+  describe('When initializing the file cache manager', () => {
+    let sut: FileCacheManager = fileCacheManagerFactory();
+    beforeEach(() => {
+      sut = fileCacheManagerFactory();
+      jest.restoreAllMocks();
+    });
+    it('Should fail if the cache manager is not initialized', () => {
+      sut.cacheFile('/non_cached', 'hi.png').catch((err) => {
+        expect(err).toBeTruthy();
+      });
+    });
+
+    it('Should fail if the directory does not exists', async () => {
+      mockedFs.exists.mockImplementationOnce(async () => false);
+      mockedFs.getDirSize.mockImplementationOnce(async () => 1);
+      sut.init().catch((err) => expect(err).toBeTruthy());
```

Comment:
is better to match it against the expected error, not just assume that the error thrown is the desired error. 

Therefore, create a class for each custom error of the cache, then check if the error is an instance of the expected error.